src/hooks/useDynamicSafeAction.ts

```diff
@@ -0,0 +1,60 @@
+import { UseMutateAsyncFunction, UseMutateFunction } from '@tanstack/react-query'
+import { SafeClientResult } from '@safe-global/sdk-starter-kit'
+import { ConfigParam, SafeConfigWithSigner } from '@/types/index.js'
+import {
+  useSendSafeOperation,
+  UseSendSafeOperationReturnType
+} from '@/hooks/useSendSafeOperation.js'
+import { useSendTransaction, UseSendTransactionReturnType } from '@/hooks/useSendTransaction.js'
+import {
+  useConfirmSafeOperation,
+  UseConfirmSafeOperationReturnType
+} from '@/hooks/useConfirmSafeOperation.js'
+import {
+  useConfirmTransaction,
+  UseConfirmTransactionReturnType
+} from '@/hooks/useConfirmTransaction.js'
+import { useConfig } from './useConfig.js'
+
+export type UseDynamicSafeActionParams = ConfigParam<SafeConfigWithSigner>
+
+export type UseDynamicSafeActionReturnType = {
+  send: UseMutateFunction<SafeClientResult, Error, any, unknown>
+  sendAsync: UseMutateAsyncFunction<SafeClientResult, Error, any, unknown>
```

Comment:
Should those `any` be `SendTransactionVariables | SendSafeOperationVariables`

Reply:
fixed